File: server/MotoRevApi/Model/Moto.cs
namespace MotoRevApi.Model;

public class Moto
{
    public int Id { get; set; }
    public required string Placa { get; set; }
    public required string Chassi { get; set; }

    public int ModeloMotoId { get; set; }
    public virtual ModeloMoto ModeloMoto { get; set; } = null!;

    public int ClienteId { get; set; }
    public virtual Cliente Cliente { get; set; } = null!;

    public int? ConcessionariaId { get; set; }
    public virtual Concessionaria? Concessionaria { get; set; }

    public bool Ativo { get; set; } = true;
    public string? Foto { get; set; }

    public string Cor { get; set; } = null!;
    public int KilometragemAtual { get; set; }
    public DateTime DataVenda { get; set; }

    public virtual ICollection<RevisaoMoto> RevisoesPlanejadas { get; set; } = new List<RevisaoMoto>();
}

File: client/src/app/models/Moto.ts
import { RevisaoPadraoPecaResponse } from '@/app/models/RevisaoPadrao';
import { Servico } from '@/app/models/Servico';

export interface RevisaoMotoResponse {
  id: number;
  revisaoPadraoId: number;
  nome: string;
  ordem: number;
  quilometragem: number;
  tempoMeses: number;
  dataPrevista: string;
  status: string;
  servicos: Servico[];
  pecas: RevisaoPadraoPecaResponse[];
}

export interface Moto {
  id: number;
  placa: string;
  chassi: string;
  modeloMotoId: number;
  nomeModelo: string;
  marca: string;
  clienteId: number;
  concessionariaId?: number;
  nomeConcessionaria?: string;
  foto?: string;
  cor: string;
  kilometragemAtual: number;
  dataVenda: string;
  linha: string;
  cilindrada: string;
  ano: number;
  revisoesPlanejadas: RevisaoMotoResponse[];
}
